### PR TITLE
Optimizing the performance of CoveragePublisher for all scenarios

### PR DESCRIPTION
**Pull request by vinayakmsft**
**Task name**: PublishCodeCoverageResults V2

**Description**: Publish Code Coverage Results V2 task uses the CoveragePublisher package. We use the CovereagePublisher.Console.exe which can perform all the tasks of the PublishCodeCoverageV2 task after the .net core upgrade. It is a selft contained file and is optimized. This exe runs without having to install the .net core sdk versions. There is no need to have the .dll file mentioned in the Coveragepublisher.ts file. 

Similarly we enabled the task for Linux and macos agent. So now the PCCRV2 task will be supported on the Linux and macos agents as well without having to install .net core sdk tasks. 

**Documentation changes required:** (Y/N) Y.

Local testing: Working correctly as expected. 

Few pipelines tests on linux, windows and macos agents covering all the scenarios:
1) Windows agents:
[Pipelines - Run 204 (azure.com)](https://dev.azure.com/ForCjson/cjsontask/_build/results?buildId=204&view=results)

2) Macos agent - https://dev.azure.com/ForCjson/cjsontask/_build/results?buildId=210&view=results

3) Linux/ubuntu agent - https://dev.azure.com/ForCjson/cjsontask/_build/results?buildId=211&view=results

Pipeline for testing:
image

Results:

image

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Review by jakubch1 — CHANGES_REQUESTED** [review]

File: Tasks/Common/coveragepublisher/coveragepublisher.ts [hunk 1]
```diff
@@ -28,14 +28,9 @@ async function publishCoverage(inputFiles: string[], reportDirectory: string, pa
...
-    if (!dotnetPath && osvar === 'win32') {
-        // use full .NET to execute
-        dotnet = taskLib.tool(path.join(__dirname, 'CoveragePublisher', 'CoveragePublisher.Console.exe'));
-    } else {
-        dotnet = taskLib.tool(dotnetPath);
-        dotnet.arg(path.join(__dirname, "CoveragePublisher", 'CoveragePublisher.Console.dll'));
-    }
-
+    // use full .NET to execute
+    dotnet = taskLib.tool(path.join(__dirname, 'CoveragePublisher', 'CoveragePublisher.Console.exe'));
```
[the remark this answers is not in the document]
Comment (vinayakmsft) [reply Tasks/Common/coveragepublisher/coveragepublisher.ts]: yes added for other agents like linux and macos

**Comment by adityashahms** [comment]
Approving based on Abhishek's comment.

**Comment by adityashahms** [comment]
Approving based on Abhishek's comment

**Review by adityashahms — APPROVED** [review]
Approving based on Abhishek's Comments

**Comment by chenlish** [comment]
@vinayakmsft Do we have a plan to deploy 223 version?

**Comment by vinayakmsft** [comment]
Hi Chen,

The PR is merged , and the sprint deployments (ring wise) will begin from next to next week. So it would  take 2-3 weeks for complete deployment.

**Comment by chenlish** [comment]
Thanks for quick reply @vinayakmsft , and any idea or workaroud before the deployment?

**Pull request merged**